core/transcoder.go

```diff
@@ -131,32 +146,30 @@ func (n *LivepeerNode) getSegmentChan(job *ethTypes.Job) (SegmentChan, error) {
 	return sc, nil
 }
 
-func (n *LivepeerNode) TranscodeSegment(job *ethTypes.Job, ss *SignedSegment) error {
+func (n *LivepeerNode) TranscodeSegment(job *ethTypes.Job, ss *SignedSegment) (*TranscodeResult, error) {
 	glog.V(common.DEBUG).Infof("Starting to transcode segment %v", ss.Seg.SeqNo)
 	ch, err := n.getSegmentChan(job)
 	if err != nil {
 		glog.Error("Could not find segment chan ", err)
-		return err
+		return nil, err
 	}
-	segChan := &SegChanData{seg: ss, res: make(chan error, 1)}
+	segChan := &SegChanData{seg: ss, res: make(chan *TranscodeResult, 1)}
 	select {
 	case ch <- segChan:
 		glog.V(common.DEBUG).Info("Submitted segment to transcode loop")
 	default:
 		// sending segChan should not block; if it does, the channel is busy
 		glog.Error("Transcoder was busy with a previous segment!")
-		return fmt.Errorf("TranscoderBusy")
-	}
-	select {
-	case err := <-segChan.res:
-		if err != nil {
-			return err
-		}
+		return nil, fmt.Errorf("TranscoderBusy")
 	}
-	return nil
+	res := <-segChan.res
+	return res, res.Err
 }
 
-func (n *LivepeerNode) transcodeAndBroadcastSeg(seg *stream.HLSSegment, sig []byte, cm eth.ClaimManager, t transcoder.Transcoder, resultStrmIDs []StreamID, config net.TranscodeConfig) error {
+func (n *LivepeerNode) transcodeAndBroadcastSeg(config transcodeConfig, ss *SignedSegment) *TranscodeResult {
```

Comment:
Even though we are putting the transcoded video into the `VideoCache`, we are no longer "broadcasting" here.  Should we rename this to `transcodeAndCacheSeg`?

Reply:
Fixed https://github.com/livepeer/go-livepeer/pull/512/commits/25e7e8b85b9038aee67a242f037d9794cf26d9d0